src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,197 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = tail = new Node<>(null, value, null);
+        } else {
+            tail = tail.next = new Node<>(tail, value, null);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        addByIndexRoutine(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkSetIndexInBound(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        return setValueRoutine(value, index);
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unlinkNodeByIndex(index);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        return unlinkNodeByObject(object);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return head == null;
+    }
+
+    private void addByIndexRoutine(T value, int index) {
+        checkIndexInBound(index);
+        if (head == null) {
+            head = tail = new Node<>(null, value, null);
+        } else if (index == 0) {
+            head = new Node<>(null, value, head);
+        } else if (index == size) {
+            tail = tail.next = new Node<>(tail, value, null);
+        } else {
+            Node<T> currentNode = getNode(index - 1);
+            Node<T> node = new Node<>(currentNode, value, currentNode.next);
+            currentNode.next.prev = node;
+            currentNode.next = node;
+
+        }
+        size++;
+    }
+
+    private T setValueRoutine(T value, int index) {
+        checkSetIndexInBound(index);
+        Node<T> nodeToSet;
+        if (index == 0) {
+            nodeToSet = head;
```

Comment:
I do not think it is necessary, `getNode()` should be able to handle such a case, if not - we should fix it there